### DISCUSSION TITLE
How to use MoshiConverterFactory for generic type of model classes in android ?

### DISCUSSION THREAD

**Discussion by GopalLR** [square/retrofit#3933] [General]
I'm working in one project which has morethen 50 APIs and all APIs response structure is same like below :

{

"status": 200,

"message": "Login successfully.",

"result": {

}

}

I create base model class with generic type

@JsonClass(generateAdapter = true)

data class BaseModel<T>(

@Json(name = "status")

val status: Int,

@Json(name = "message")

val message: String,

@Json(name = "result")

val result: Map<String, T>
)

@JsonClass(generateAdapter = true)

data class AccessToken(

@Json(name = "accessToken")

val accessToken: String
)

It's give error like :

Unable to create converter for com.app.sample.data.model.BaseModel<com.app.sample.data.model.AccessToken> for method AuthService.loginUser in android

How to fixed this issue using MoshiConverterFactory ?

**Comment by JakeWharton** [comment]
I moved back to Moshi discussions. See https://github.com/square/moshi/discussions/1721?converting=1#discussioncomment-6730475
